R/extract.R

```diff
@@ -82,13 +82,10 @@ get_chunk_positions <- function(pattern, lines) {
   #   set the initial column to the leftmost one within each chunk (including the start+end gates). See tests.
   # use 'ws_re' to make clear that we're matching knitr's definition of initial whitespace.
   ws_re <- sub("```.*", "", pattern$chunk.begin)
-  extract_indent <- function(start, end) {
-    matches <- gregexpr(ws_re, lines[start:end], perl = TRUE)
-    lengths <- vapply(matches, attr, integer(1L), "match.length")
-    min(lengths)
-  }
-  # min() guarantees length(indents) == length(starts)
-  indents <- unlist(Map(extract_indent, starts, ends))
+  # NB: min() guarantees length(indents) == length(starts)
+  indents <- unlist(
+    Map(function(start, end) min(attr(regexpr(ws_re, lines[start:end], perl = TRUE), "match.length")), starts, ends)
```

Comment:
I personally find the inlined version much harder to read. Extracting and naming the function makes the code here much easier to follow IMO.